app/scripts/lib/account-tracker.js

```diff
@@ -70,22 +66,27 @@ export default class AccountTracker {
     };
 
     this._provider = opts.provider;
-    this._query = pify(new EthQuery(this._provider));
     this._blockTracker = opts.blockTracker;
-    // blockTracker.currentBlock may be null
-    this._currentBlockNumber = this._blockTracker.getCurrentBlock();
-    this._blockTracker.once('latest', (blockNumber) => {
-      this._currentBlockNumber = blockNumber;
-    });
+
     // bind function for easier listener syntax
     this._updateForBlock = this._updateForBlock.bind(this);
     this.getCurrentChainId = opts.getCurrentChainId;
+    this.getNetworkClientById = opts.getNetworkClientById;
     this.getNetworkIdentifier = opts.getNetworkIdentifier;
```

Comment:
tangent question: can we remove `getNetworkIdentifier`?

Reply:
Sure, we'd have to bring in this logic https://github.com/MetaMask/metamask-extension/blob/e72878f04f85ff4e6e4b8ffb96848b947dad1238/app/scripts/metamask-controller.js#L878C5-L878C5

Reply:
any reason not to?

Reply:
keeping changes as minimal as possible to the task at hand. I'll make those changes in this PR though. I to want to avoid doing cleanup for each controller we touch since I've seen countless things that can be improved but aren't necessary to change in order to add networkClientId support

Reply:
Yes you're right. We shouldn't do this in this PR. I didn't mean to imply we should do it in this PR. I just noticed it and was wondering since we'd removed the networkId elsewhere.

Reply:
Ohh. This isn't networkId. In this context, this actually is just the RPC URL if the network type is custom

Reply:
oh thats super confusing lol. We should rename this at some point!

Reply:
💯 

there's another reference or two floating around. Should probably addresses them all in one go